core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -655,7 +656,16 @@ public void setDefaultTTL(int defaultTTL) {
                         .doOnNext(i -> insertedDataPointEvents.onNext(metric)))
                 .doOnNext(meter::mark);
 
-        Observable<Integer> indexUpdates = dataAccess.updateMetricsIndex(metrics)
+        Observable<Metric<T>> updateIdxMetrics = metrics.filter(metric -> {
```

Comment:
Not sure if this is a good idea because this will make at least the outcome of some tests non-deterministic, since the metric will not be retrievable. Also, we reduce the metrics idx updates by a lot with this caching strategy, so I would like to keep the guarantee that a metric is retrievable immediately after insertion of data point.